src/main/java/net/sf/jabref/logic/exporter/CustomExportList.java

```diff
@@ -93,20 +99,22 @@ private void readPrefs(JabRefPreferences prefs) {
         } else {
             lfFileName = s.get(1);
         }
-        ExportFormat format = new ExportFormat(s.get(0), s.get(0), lfFileName, null, s.get(2));
+        ExportFormat format = new ExportFormat(s.get(0), s.get(0), lfFileName, null, s.get(2), layoutPreferences,
+                savePreferences);
         format.setCustomExport(true);
         return Optional.of(format);
     }
 
-    public void addFormat(List<String> s) {
-        createFormat(s).ifPresent(format -> {
+    public void addFormat(List<String> s, LayoutFormatterPreferences layoutPreferences, SavePreferences savePreferences) {
+        createFormat(s, layoutPreferences, savePreferences).ifPresent(format -> {
             formats.put(format.getConsoleName(), format);
             list.add(s);
         });
     }
 
-    public void remove(List<String> toRemove) {
-        createFormat(toRemove).ifPresent(format -> {
+    public void remove(List<String> toRemove, LayoutFormatterPreferences layoutPreferences,
```

Comment:
There are indeed many injections of preferences in different ways. I'm not really sure that this is the best approach (or rather sure that there is probably a better). In addition, the only instance of CustomExportList exists in JabRefPreferences... However, at the moment I was more focused on getting rid of Globals.prefs. Regarding granularity, it is always tricky.